AmbientLightSensor: add some basic functionality tests

Honry opened the pull request:
Cover start(), stop(), onchange, onstatechange and also secure stuff when browsing context

zqzhang commented on ambient-light/AmbientLightSensor_start.html:
```diff
@@ -0,0 +1,35 @@
+});
+
+test(function() {
+  assert_equals(sensor.state, "activating");
```
I think it is better to put all these assertions in this test file together within `ambient-light/AmbientLightSensor_onstatechange.html`.

Honry replied:
@zqzhang, thanks for the comments, do you mean sub-case is unnecessary here?

Honry replied:
@zqzhang, thanks for your comments! Agree!

zqzhang commented on ambient-light/AmbientLightSensor_stop.html:
```diff
@@ -0,0 +1,34 @@
+});
+
+test(function() {
+  assert_equals(sensor.state, "idle");
```
I think it is better to put all these assertions in this test file together within `ambient-light/AmbientLightSensor_onstatechange.html`.

zqzhang commented:
@Honry I mean it is better to combine all the tests in `AmbientLightSensor_{onchange|start|stop}.html` into one test file, and to check things in each state, e.g. before and after start, before and after stop. Then the duplicates go away by nature.

Honry committed:
AmbientLightSensor: add some basic functionality tests

Cover start(), stop(), onchange, onstatechange and secure stuff when browsing context

Honry commented:
@zqzhang, initially I designed 3 test files for onstatechange, start, and stop since they may be extended with more checkpoints in future, such as permission, 'active' state, reporting mode and etc., to make each test file distinct. WDYT?

Honry committed:
Remove duplicate assertions for sensor.state

Which has been covered in AmbientLightSensor_onstatechange.html
